### PR TITLE
Rudimentary Search

### PR DESCRIPTION
**Pull request by ajvalera**
Added a rudimentary search form under courses#index. 

Next steps are to diversify options for search.

### CODE REVIEW

**Comment by rrshaban** [comment]
PULL REQUEST NUMBER TWOOOO

File: app/views/courses/search.html.erb [hunk 1]
```diff
@@ -21,9 +18,25 @@ What you are looking for cannot be found.
         <td><%= Professor.find(course.professor_id).name %></td>
 
         <td><%= link_to 'Show', course %></td>
-        <td><%= link_to 'Edit', edit_course_path(course) %></td>
-        <td><%= link_to 'Destroy', course, method: :delete, data: { confirm: 'Are you sure?' } %></td>
       </tr>
     <% end %>
   </tbody>
 </table>
+
```
Comment (rrshaban) [inline app/views/courses/search.html.erb]: so, uh, did you think about putting this code in a partial?

File: app/views/courses/search.html.erb [hunk 2]
```diff
@@ -21,9 +18,25 @@ What you are looking for cannot be found.
...
+      </li>
+      <li>
+        <%= f.label :crn_cont %>
+        <%= f.text_field :crn_cont %>
+      </li>
+    </ul>
+  </fieldset>
+
+  <%= f.submit "Search!" %>
+<% end %>
```
Comment (rrshaban) [inline app/views/courses/search.html.erb]: we could probably stick this search code in a partial, given we're calling it a bunch from a bunch of different files. shared/_search_bar.html.erb maybe?

**Comment by rrshaban** [comment]
I moved it to a partial. Let's talk also about moving the search to a different REST route, instead of layering it on top of index/show

**Comment by ajvalera** [comment]
Okay, I was maybe thinking we could have this be the default route, and we make rudimentary search available; then, we leave courses#search for an "advanced searching" mode, a la

https://github.com/activerecord-hackery/ransack_demo

I don't know how valuable a giant list of courses by itself is, so I figured if the default is just to see all the courses, it wouldn't hurt. Maybe it's a moot point, but sure, let's talk about it.

**Comment by rrshaban** [comment]
Yeah that's fair. I was thinking for Index we could listed highest rated classes by category, new reviews, courses that have yet to be reviewed, etc.

**Pull request merged**